integrations/DCMFloodlight/utils.js

```diff
@@ -72,4 +91,224 @@ const mapFlagValue = (key, value) => {
   );
 };
 
-export { transformCustomVariable, mapFlagValue };
+const calculateQuantity = (products) => {
+  if (products && products.length > 0 && Array.isArray(products)) {
+    return products.reduce((accumulator, product) => {
+      if (product.quantity) {
+        return accumulator + product.quantity;
+      }
+      return accumulator;
+    }, 0);
+  }
+  return 0;
+};
+
+const buildGtagTrackPayload = (
+  message,
+  salesTag,
+  countingMethod,
+  integrationObj
+) => {
+  let dcCustomParams = {
+    ord: get(message, "properties.ord"),
+    dc_lat: get(message, "context.device.adTrackingEnabled"),
+  };
+
+  let eventSnippetPayload = {};
+  if (salesTag) {
+    // sales tag
+    dcCustomParams.ord =
+      get(message, "properties.orderId") || get(message, "properties.order_id");
+    let qty = get(message, "properties.quantity");
+
+    eventSnippetPayload = {
+      ...eventSnippetPayload,
+      value: get(message, "properties.revenue"),
+      transaction_id: dcCustomParams.ord,
+    };
+
+    // sums quantity from products array or fallback to properties.quantity
+    const products = get(message, "properties.products");
+    const quantities = calculateQuantity(products);
+    if (quantities) {
+      qty = quantities;
+    }
+
+    // Ref - https://support.google.com/campaignmanager/answer/7554821#zippy=%2Cfields-in-event-snippets-for-sales-tags
+    switch (countingMethod) {
+      case "transactions":
+        break;
+      case "items_sold":
+        if (qty) {
+          eventSnippetPayload.quantity = parseFloat(qty);
+        }
+        break;
+      default:
+        logger.error("[DCM Floodlight] Sales Tag:: invalid counting method");
+        return;
+    }
+  } else {
+    // counter tag
+    // Ref - https://support.google.com/campaignmanager/answer/7554821#zippy=%2Cfields-in-event-snippets-for-counter-tags
+    switch (countingMethod) {
+      case "standard":
+        break;
+      case "unique":
+        dcCustomParams.num = get(message, "properties.num");
+        break;
+      case "per_session":
+        dcCustomParams.ord = get(message, "properties.sessionId");
+        eventSnippetPayload.session_id = dcCustomParams.ord;
```

Comment:
do we need to set sessionId to ord? Seems like this is for session identification